clients/include/testing_axpy.hpp

```diff
@@ -17,6 +17,15 @@
 
 using namespace std;
 
+#define CLEANUP()                                                  \
```

Comment:
Code like this would benefit from the application of std::unique_ptr<>.  Then the CLEANUP macro goes away, and you don't have to worry about calling CLEANUP right before every return.